agent/acs/handler/acs_handler.go

```diff
@@ -281,17 +281,18 @@ func (acsSession *session) startACSSession(client wsclient.ClientServer) error {
 		acsSession.addUpdateRequestHandlers(client)
 	}
 
-	err := client.Connect()
+	disconnectTimer, err := client.Connect(metrics.ACSDisconnectTimeoutMetricName,
+		acsSession.connectionTime,
+		acsSession.connectionJitter)
 	if err != nil {
 		seelog.Errorf("Error connecting to ACS: %v", err)
 		return err
 	}
+	if disconnectTimer != nil {
```

Comment:
Is this check necessary? My understanding is that this shouldn't ever be nil if err == nil (which we assert/check on L287).

Reply:
Its actually failing for a test, where we return nil and nil, because we do not create a timer in gomock. Let me go back and check if we can fix the test.